src/sinks/prometheus/exporter.rs

```diff
@@ -151,16 +152,44 @@ struct PrometheusExporter {
     server_shutdown_trigger: Option<Trigger>,
     config: PrometheusExporterConfig,
     metrics: Arc<RwLock<IndexSet<MetricEntry>>>,
-    last_flush_timestamp: Arc<RwLock<i64>>,
     acker: Acker,
 }
 
+#[derive(Debug)]
+struct MetricEntry {
+    pub metric: Metric,
+    pub next_flush: Instant,
+}
+
+impl MetricEntry {
+    fn new(metric: Metric, next_flush: Instant) -> Self {
+        Self { metric, next_flush }
+    }
+
+    fn is_expired(&self, time: Instant) -> bool {
+        self.next_flush < time
+    }
+}
+
+impl Hash for MetricEntry {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        BufferMetricEntry::metric_hash(&self.metric, state)
```

Comment:
Can this not use the `hash` method directly?
```suggestion
        self.metric.hash(state)
```